src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,153 @@
+                                SSLPort = port;
+                                break;
+                            case var _ when key.SequenceEqual(nameof(NonSSLPort).ToCharArray()) && Int32.TryParse(value.ToString(), out var nonsslport):
+                                NonSSLPort = nonsslport;
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+            catch
+            {
+                // TODO: Append to rolling debug log when it's present
+            }
+        }
+
+        internal async static Task AddListenerAsync(ConnectionMultiplexer multiplexer, LogProxy logProxy)
+        {
+            try
+            {
+                var sub = multiplexer.GetSubscriber();
+                if (sub == null)
+                {
+                    logProxy?.WriteLine("Failed to GetSubscriber for AzureRedisEvents");
+                    return;
+                }
+
+                await sub.SubscribeAsync(PubSubChannelName, (channel, message) =>
+                {
+                    var newMessage = new AzureMaintenanceEvent(message);
+                    multiplexer.InvokeServerMaintenanceEvent(newMessage);
+
+                    if (newMessage.NotificationType.Equals("NodeMaintenanceEnded") || newMessage.NotificationType.Equals("NodeMaintenanceFailover"))
+                    {
+                        multiplexer.ReconfigureAsync(first: false, reconfigureAll: true, log: logProxy, blame: null, cause: $"Azure Event: {newMessage.NotificationType}").Wait();
+                    }
+                }).ForAwait();
+            }
+            catch (Exception e)
+            {
+                logProxy?.WriteLine($"Encountered exception: {e}");
+            }
+        }
+
+        /// <summary>
+        /// Raw message received from the server
+        /// </summary>
+        public readonly string RawMessage;
+
+        /// <summary>
+        /// indicates the event type
+        /// </summary>
+        public readonly string NotificationType;
+
+        /// <summary>
+        /// indicates the start time of the event
+        /// </summary>
+        public readonly DateTime? StartTimeInUTC;
+
+        /// <summary>
+        /// indicates if the event is for a replica node
+        /// </summary>
+        public readonly bool IsReplica;
+
+        /// <summary>
+        /// IPAddress of the node event is intended for
+        /// </summary>
+        public readonly IPAddress IpAddress;
+
+        /// <summary>
+        /// ssl port
+        /// </summary>
+        public readonly int SSLPort;
+
+        /// <summary>
+        /// non-ssl port
+        /// </summary>
+        public readonly int NonSSLPort;
+
+        /// <summary>
+        /// Returns a string representing the maintenance event with all of its properties
+        /// </summary>
+        public override string ToString()
```

Comment:
I wonder if this should just return `RawMessage` ?

Reply:
👍 will update